modules/data.atmosphere/R/solar_angle.R

```diff
@@ -2,6 +2,27 @@
 #'
 #' For explanations of formulae, see https://web.archive.org/web/20180307133425/http://www.itacanet.org/the-sun-as-a-source-of-energy/part-3-calculating-solar-angles/
 #'
+#' @description
+#' `cos_solar_zenith_angle` returns the cosine of solar zenith angle.
+#' The cosine of the solar zenith angle (cossza) is the angle between the sun’s rays and the vertical
+#' This solar zenith angle is normally used in combination with the solar azimuth angle to determine the position of the Sun as observed from a given location on the surface of the Earth.
+#' The cosine of the solar zenith angle (cossza) is also a key component in Mean Radiant Temperature(MRT).
+#' It is required for photolysis.
+#'
+#' doy Day of year : It is the day when you are calculating the angle.
+#' lat Latitude : Latitude measures the distance north or south of the equator. The primary unit of latitude is degrees (°).
+#' lon Longitude : Longitude measures distance east or west of the prime meridian. The primary unit of longitude is degrees (°).
+#' dt Timestep : Timestep is a scalar quantity.
+#' hr Hours timestep : It is a timestep in hours format.
```

Comment:
Still doesn't really explain what dt and hr actually are and what acceptable values are. Looking at the code, hr isn't a "timestep" it's the time of day that you're trying to predict zenith angle for (If I'd written this function I'd have put it next to doy to make this connection more obvious). Looking at the code I'd guess this should be a number between 0-24 and that decimal values are acceptable (i.e. one should use fractional hours instead of minutes). I agree dt is probably a scalar, but you likewise still haven't said what it actually is or what acceptable values are. My best guess it that the author of the function anticipated that many people would be passing in vectors of days and hours, and thus if you have a whole vector of times you might actually want to do the calculation for the midpoint between two timesteps. dt thus tells the function how far apart values in hr are. Seeing that this is calculated by dividing dt by 86400 and then multiplying by 24, it looks like dt needs to be in SECONDS. In my mind dt should probably be given a default of 0.

Reply:
sure, I have made the changes in the code and made it more in detail. I have one doubt should I write an example, if so then it'll take too many lines.